Documentation/building/crossgen.md

```diff
@@ -0,0 +1,64 @@
+Using CrossGen to Create Native Images
+======================================
+
+Introduction
+------------
+
+When you create a .NET assembly using C# compiler, your assembly contains only MSIL code.
+When the app runs, the JIT compiler translates the MSIL code into native code, before the CPU can execute them.
+This execution model has some advantages. For example, your assembly code can be portable across all platforms and architectures that supports .NET Core.
+However, this portability comes with a performance penalty. Your app starts up slower, because the JIT compiler has to spend time to translate the code.
+
+To help make your app start up faster, CoreCLR includes a tool called crossgen, which can pre-compile the MSIL code into native code.
+
+Getting crossgen
+----------------
+
+If you build CoreCLR yourself, the crossgen tool (`crossgen.exe` on Windows, or `crossgen` on other platforms) is created as part of the build, and stored in the same output directory as other CoreCLR binaries.
+If you install CoreCLR using a NuGet package, you can find crossgen in the `tools` folder of the NuGet package.
+
+Regardless of how you obtain crossgen, it is very important that it must match other CoreCLR binaries.
+- If you build crossgen yourself, you should use it with coreclr and mscorlib generated from the same build. Do not attempt to mix crossgen from one build with binaries generated from another build.
+- If you install crossgen from NuGet, make sure you use crossgen from exactly the same NuGet package as the rest of you CoreCLR binaries. Do not attempt to mix binaries from multiple NuGet packages.
+
+If you do not follow the above rules, you are likely to encounter errors while running crossgen.
+
+Using crossgen
+--------------
+
+In most cases, the build script automatically runs crossgen to create the native image for mscorlib.
+When this happens, you will find boths `mscorlib.dll` and `mscorlib.ni.dll` in your output directory.
+`mscorlib.dll` is the MSIL assembly creating by the C# compiler, while `mscorlib.ni.dll` is the native image that contains CPU-specific code.
```

Comment:
Typo. 
"...is the MSIL assembly creating ..." 
should be 
"...is the MSIL assembly created ..."